### PR TITLE
set allow any as the default for outgoing traffic

### PR DESCRIPTION
**Pull request by linsun**
Per env WG discussion, fixes: https://github.com/istio/istio/issues/11739

also disabled egress gateway by default, but enabled egress gateway for demo.

### CODE REVIEW

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -380,12 +382,12 @@ global:
   trustDomain: ""
 
   # Set the default behavior of the sidecar for handling outbound traffic from the application:
-  # REGISTRY_ONLY - restrict outbound traffic to services defined in the service registry as well
-  #   as those defined through ServiceEntries
   # ALLOW_ANY - outbound traffic to unknown destinations will be allowed, in case there are no
```
Comment (costinm) [inline install/kubernetes/helm/istio/values.yaml]: Some indication it is the default ?

**Comment by costinm** [comment]
Also for egress gateway - more comments please...
"If you are using egress gateway rules ( link to doc ?) you will need to update istio and set this option to true. Depending on expected egress traffic, you should also adjust CPU request and max autoscale replicas for the egress, etc"

File: install/kubernetes/helm/istio/text.yaml [hunk 2]
```diff
@@ -0,0 +1,17209 @@
+---
+# Source: istio/charts/galley/templates/poddisruptionbudget.yaml
```
Comment (rshriram) [inline install/kubernetes/helm/istio/text.yaml]: stray file

File: install/kubernetes/helm/istio/values-istio-demo.yaml [hunk 3]
```diff
@@ -6,9 +6,6 @@ global:
     accessLogFile: "/dev/stdout"
 
   disablePolicyChecks: false
-  
-  outboundTrafficPolicy:
-    mode: ALLOW_ANY
```
Comment (rshriram) [inline install/kubernetes/helm/istio/values-istio-demo.yaml]: please keep this explicit

**Comment by rshriram** [comment]
what about breaking changes in behavior since past versions had registry_only behavior

File: install/kubernetes/helm/istio/text.yaml (hunk 2 above)
Comment (linsun) [reply install/kubernetes/helm/istio/text.yaml]: oops thanks!

**Comment by linsun** [comment]
@rshriram because there is a lot of breaking changes in 1.1, we feel it is okay to do the right thing for our users.   We proposed to have a flag like backwardcompatible in helm to guard these backwards compatible options which seems favorable by folks from env WG, I can add that in a follow up PR.

File: install/kubernetes/helm/istio/values.yaml (hunk 1 above)
Comment (linsun) [reply install/kubernetes/helm/istio/values.yaml]: done!

File: Makefile [hunk 4]
```diff
@@ -680,6 +682,8 @@ generate_yaml: $(HELM) $(HOME)/.helm istio-init.yaml
 		--set global.controlPlaneSecurityEnabled=false \
 		--set global.proxy.enableCoreDump=${ENABLE_COREDUMP} \
 		--set istio_cni.enabled=${ENABLE_ISTIO_CNI} \
+		--set gateways.istio-egressgateway.enabled=true \
+		--set global.outboundTrafficPolicy.mode=REGISTRY_ONLY \
```
Comment (ymesika) [inline Makefile]: remove space at the end of this line

Comment (linsun) [reply Makefile]: thanks!!

**Comment by ymesika** [comment]
cc @wenchenglu

**Comment by linsun** [comment]
@wenchenglu can we get this in 1.1?  This change is approved by env WG yesterday

**Comment by linsun** [comment]
a little fun fact, I got a lot of support on this:

https://twitter.com/linsun_unc/status/1098429168025550848

**Comment by ymesika** [comment]
@linsun Egress is one of the major pain points and source for frustration with novice users imho.
Good move.

**Pull request merged**